tests/samplers_tests/test_samplers.py

```diff
@@ -84,6 +86,51 @@
         else [lambda: optuna.integration.BoTorchSampler(n_startup_trials=0)]
     ),
 )
+parametrize_reseedable_sampler = copy.deepcopy(parametrize_sampler)
+parametrize_reseedable_sampler.args[1].append(
+    lambda: PartialFixedSampler(
+        fixed_params={"x": 0}, base_sampler=optuna.samplers.RandomSampler()
+    )
+)
+
+
+@parametrize_reseedable_sampler
+def test_sampler_reseed_rng(sampler_class: Callable[[], BaseSampler]) -> None:
+    sampler = sampler_class()
+    has_rng = "_rng" in sampler.__dict__
```

Comment:
This depends on the name of each sampler's private attribute, which I think is too much of a dependency on the implementation. How about checking each `attr` of `sampler.__dict__` is an instance of `np.random.RandomState`?

Reply:
Thank you for your proposal. I agree with you.

Reply:
done: https://github.com/optuna/optuna/pull/3569/commits/f1845b1356df40621ceca2a7fa127a70dabe424a